src/SFA.DAS.ApprenticePortal/Application/Extensions/TaskEx.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticePortal.Application.Extensions
+{
+    public static class TaskEx
+    {
+        public static async Task<(T1, T2)> AwaitAll<T1, T2>(Task<T1> t1, Task<T2> t2)
+        {
+            await Task.WhenAll(t1, t2);
+            return (t1.Result, t2.Result);
+        }
+    }
+}
```

Comment:
My only 2 reservations against this pattern is 

1. it's actually quite hard to debug, if one of the tasks is failing. You have to refactor the code to determine which task is failing.
2. What happens when 3 or more tasks can be executed at the same time, be better as an input array, but that's a minor point

Reply:
you would need to put a break point in the inner api calls or try catch to debug.